vm/src/obj/objset.rs

```diff
@@ -106,76 +106,31 @@ impl PySetInner {
         self.content.contains(vm, needle)
     }
 
-    #[inline]
-    fn _compare_inner(
+    fn compare(
         &self,
         other: &PySetInner,
-        size_func: fn(usize, usize) -> bool,
-        swap: bool,
+        op: PyComparisonOp,
         vm: &VirtualMachine,
     ) -> PyResult<bool> {
-        let (zelf, other) = if swap { (other, self) } else { (self, other) };
-
-        if size_func(zelf.len(), other.len()) {
+        if op == PyComparisonOp::Ne {
+            return self.compare(other, PyComparisonOp::Eq, vm).map(|eq| !eq);
+        }
```

Comment:
is this still required? It seems eq and ne can be handled by eval_ord-like way.

Reply:
Adding this fixed an issue, I think it's kinda a weird case cause for `ne` and only `ne`, if a comparison that's predicated on equality (like `bool_seq_{lt,gt}`) fails, then it should return true instead of false. I'll try again to figure out how to make it work without this special-case, I agree that it seems like it should be possible.